### DISCUSSION TITLE
Hue Motion Node - Topic, Brightness, Timer

### DISCUSSION THREAD

**Discussion by Trinta3** [Supergiovane/node-red-contrib-knx-ultimate#385] [Ideas]
Hello! 

First, I'd like to thank you for your amazing work! I was already impressed by KNX Ultimate, but when I saw you develop the Hue integration, wow!

I'm trying to use the Hue Motion Node to switch on/off a LED stripe in my KNX environment. The Motion detector is already configured in Hue to control a group of 3 Hue Lights, but I'd like to add the control of that LED stripe to it.

Initially, I thought the Node would use my existing Hue app configuration, but I noticed that my LED switched on even during the day, while the Hue lights stayed off. As I'm not a power user of Node-RED, I might be using it incorrectly.

I'm now trying to configure some parameters manually in Node-RED and noticed that:
- The Hue Motion node doesn't allow to add a specific topic like the KNX Device nodes.
- It would be helpful to have a timer integrated into the node.
- It would be useful to be able to set a minimum brightness to activate the motion.

Again, I'm not sure what's happening behind the scenes in your code or if these features are even possible. Thank you again for everything (a donation has been done).

**Comment by Supergiovane** [comment]
Hi
The HUE motion sensor devices reports light lux and motion events to the HUE bridge, that compare the two and turn on and off the light.
In node-red you must compare the values by yourself. Please see this example.

<img width="1221" alt="Node-RED___Flow_3" src="https://github.com/user-attachments/assets/234ba287-9822-4f68-9d37-a46f36c6602d">

 Import this code to your flow and modify all the thing you want.
`
[
    {
        "id": "01d1b419300f9694",
        "type": "knxUltimateHueMotion",
        "z": "9161ce8b475008f7",
        "server": "058bf6746b8087d4",
        "serverHue": "01557a6fb93bc361",
        "name": "Motion: Movimento finestrone scale",
        "namemotion": "",
        "GAmotion": "",
        "dptmotion": "",
        "hueDevice": "cbdef2c0-987e-40b3-bbf1-fbbe8b1cb23c",
        "x": 180,
        "y": 140,
        "wires": [
            [
                "b9e7888642dfb83a"
            ]
        ]
    },
    {
        "id": "dc43cb1876515222",
        "type": "knxUltimateHueLightSensor",
        "z": "9161ce8b475008f7",
        "server": "058bf6746b8087d4",
        "serverHue": "01557a6fb93bc361",
        "name": "Light Level: Movimento finestrone scale",
        "namelightsensor": "",
        "GAlightsensor": "",
        "dptlightsensor": "",
        "readStatusAtStartup": "yes",
        "hueDevice": "9a75467c-4131-45d4-bba6-29cbae30efc8",
        "x": 190,
        "y": 240,
        "wires": [
            [
                "66f5a929265d9fd9"
            ]
        ]
    },
    {
        "id": "7e38330516669ad1",
        "type": "knxUltimate",
        "z": "9161ce8b475008f7",
        "topic": "",
        "setTopicType": "str",
        "outputtopic": "",
        "dpt": "",
        "initialread": 0,
        "notifyreadrequest": false,
        "notifyresponse": false,
        "notifywrite": true,
        "notifyreadrequestalsorespondtobus": false,
        "notifyreadrequestalsorespondtobusdefaultvalueifnotinitialized": "0",
        "name": "",
        "outputtype": "write",
        "outputRBE": "true",
        "inputRBE": "false",
        "formatmultiplyvalue": 1,
        "formatnegativevalue": "leave",
        "formatdecimalsvalue": 999,
        "passthrough": "no",
        "sendMsgToKNXCode": "",
        "receiveMsgFromKNXCode": "",
        "listenallga": "",
        "x": 1030,
        "y": 180,
        "wires": [
            []
        ]
    },
    {
        "id": "66f5a929265d9fd9",
        "type": "switch",
        "z": "9161ce8b475008f7",
        "name": "",
        "property": "payload",
        "propertyType": "msg",
        "rules": [
            {
                "t": "gte",
                "v": "500",
                "vt": "str"
            },
            {
                "t": "else"
            }
        ],
        "checkall": "true",
        "repair": false,
        "outputs": 2,
        "x": 430,
        "y": 240,
        "wires": [
            [
                "13cf596419321abb"
            ],
            [
                "440f8c8387d0b383"
            ]
        ]
    },
    {
        "id": "13cf596419321abb",
        "type": "function",
        "z": "9161ce8b475008f7",
        "name": "true",
        "func": "msg.topic = \"lightSensor\"\nmsg.payload = true;\nreturn msg;",
        "outputs": 1,
        "timeout": 0,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 570,
        "y": 240,
        "wires": [
            [
                "a1b17c7823d24b11"
            ]
        ]
    },
    {
        "id": "440f8c8387d0b383",
        "type": "function",
        "z": "9161ce8b475008f7",
        "name": "false",
        "func": "msg.topic = \"lightSensor\"\nmsg.payload = false;\nreturn msg;",
        "outputs": 1,
        "timeout": 0,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 570,
        "y": 280,
        "wires": [
            [
                "a1b17c7823d24b11"
            ]
        ]
    },
    {
        "id": "b9e7888642dfb83a",
        "type": "function",
        "z": "9161ce8b475008f7",
        "name": "Set Topic",
        "func": "msg.topic = \"motion\"\nreturn msg;",
        "outputs": 1,
        "timeout": 0,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 560,
        "y": 140,
        "wires": [
            [
                "a1b17c7823d24b11"
            ]
        ]
    },
    {
        "id": "53d7c6d169cd649a",
        "type": "comment",
        "z": "9161ce8b475008f7",
        "name": "Is the light level is > 500 lux?",
        "info": "",
        "x": 500,
        "y": 200,
        "wires": []
    },
    {
        "id": "a1b17c7823d24b11",
        "type": "BooleanLogicUltimate",
        "z": "9161ce8b475008f7",
        "name": "",
        "payloadPropName": "payload",
        "filtertrue": "both",
        "persist": true,
        "sInitializeWith": "WaitForPayload",
        "triggertopic": "trigger",
        "outputtriggeredby": "all",
        "inputCount": 2,
        "topic": "result",
        "restrictinputevaluation": false,
        "delayEvaluation": 0,
        "x": 780,
        "y": 180,
        "wires": [
            [
                "7e38330516669ad1"
            ],
            [],
            []
        ]
    },
    {
        "id": "058bf6746b8087d4",
        "type": "knxUltimate-config",
        "host": "224.0.23.12",
        "port": "3671",
        "physAddr": "15.15.22",
        "hostProtocol": "Auto",
        "suppressACKRequest": false,
        "csv": "",
        "KNXEthInterface": "Auto",
        "KNXEthInterfaceManuallyInput": "",
        "stopETSImportIfNoDatapoint": "fake",
        "loglevel": "info",
        "name": "KNX Gateway senza csv",
        "localEchoInTunneling": true,
        "delaybetweentelegrams": "25",
        "ignoreTelegramsWithRepeatedFlag": false,
        "autoReconnect": "yes"
    },
    {
        "id": "01557a6fb93bc361",
        "type": "hue-config",
        "host": "192.168.1.36",
        "name": "Hue Bridge",
        "bridgeid": "001788FFFE23BC7A"
    }
]
`

**Reply by Trinta3** [reply]
Thank you for your hint. 
Meanwhile I resolved this by using my KNX weather station's day/night GA.
I also managed to properly configure the function and trigger nodes to set a custom timer.
There is always room for improvement, but I am happy with the result for now.
I could gladly post in "Show off" to help others if you'd like.

![image](https://github.com/user-attachments/assets/00e621b9-3283-406a-8978-41f6b7d4eb9e)
![image](https://github.com/user-attachments/assets/3fbbb11d-3dfc-4026-a058-9ab023a15de4)

**Comment by Supergiovane** [comment]
Great. Thanks!
